source/request-as-event-emitter.js

```diff
@@ -119,23 +117,23 @@ module.exports = (options = {}) => {
 					return;
 				}
 
-				const err = new RequestError(error, options);
-				emitter.emit('retry', err, retried => {
+				if (!(error instanceof GotError)) {
```

Comment:
I wasn't sure that this was the precise change needed. My objective was to expose the TimeoutError to a client using the Promise interface.